DependencyInjection/Compiler/FirewallCompilerPass.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Scheb\TwoFactorBundle\DependencyInjection\Compiler;
+
+use Ivory\CKEditorBundle\Exception\DependencyInjectionException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\DefinitionDecorator;
+use Symfony\Component\DependencyInjection\Reference;
+
+class FirewallCompilerPass implements CompilerPassInterface
```

Comment:
This is definitely the most unstable part, because it relies on the authentication providers named `security.authentication.provider.*.*` to decorate them. If the security-bundle ever changes this, it would break.